server/service-manager.js

```diff
@@ -60,13 +60,13 @@ ServiceManager.prototype.onGatewayDestroy = onGatewayDestroy;
 
 function onProcessListen(proc, callback) {
   debug('onProcessListen(%j)', proc);
-  setImmediate(callback);
+  setImmediate(callback, null, proc);
```

Comment:
Strictly speaking this one isn't necessary since the place it was injected wasn't forwarding the result, but I think this is a safer default hook anyway.